api/src/main/java/io/opentelemetry/api/trace/SpanContext.java

```diff
@@ -66,60 +51,51 @@ private static SpanContext create(
    * @param traceState the trace state for the span context.
    * @return a new {@code SpanContext} with the given identifiers and options.
    */
-  public static SpanContext createFromRemoteParent(
+  static SpanContext createFromRemoteParent(
       String traceIdHex, String spanIdHex, byte traceFlags, TraceState traceState) {
-    return create(traceIdHex, spanIdHex, traceFlags, traceState, /* remote=*/ true);
+    return ImmutableSpanContext.create(
+        traceIdHex, spanIdHex, traceFlags, traceState, /* remote=*/ true);
   }
 
-  abstract String getTraceIdHex();
-
-  abstract String getSpanIdHex();
-
   /**
    * Returns the trace identifier associated with this {@code SpanContext}.
    *
    * @return the trace identifier associated with this {@code SpanContext}.
    */
-  public String getTraceIdAsHexString() {
-    return getTraceIdHex();
-  }
+  String getTraceIdAsHexString();
 
   /**
    * Returns the byte[] representation of the trace identifier associated with this {@link
    * SpanContext}.
    */
-  @Memoized
-  public byte[] getTraceIdBytes() {
-    return TraceId.bytesFromHex(getTraceIdHex(), 0);
+  default byte[] getTraceIdBytes() {
+    return TraceId.bytesFromHex(getTraceIdAsHexString(), 0);
   }
 
   /**
    * Returns the span identifier associated with this {@code SpanContext}.
    *
    * @return the span identifier associated with this {@code SpanContext}.
    */
-  public String getSpanIdAsHexString() {
-    return getSpanIdHex();
-  }
+  String getSpanIdAsHexString();
 
   /**
    * Returns the byte[] representation of the span identifier associated with this {@link
    * SpanContext}.
    */
-  @Memoized
-  public byte[] getSpanIdBytes() {
-    return SpanId.bytesFromHex(getSpanIdHex(), 0);
+  default byte[] getSpanIdBytes() {
+    return SpanId.bytesFromHex(getSpanIdAsHexString(), 0);
   }
```

Comment:
Why removing the ability to memoized this call? It may get expensive. I would suggest to not default this and any method that was memoized before.

Reply:
This is still memoized by the autovalue implementation. See ImmutableSpanContext above.